pkg/kubelet/server/streaming/request_cache.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package streaming
+
+import (
+	"container/list"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"k8s.io/kubernetes/pkg/util/clock"
+)
+
+var (
+	CacheTTL = 1 * time.Minute
+	// The maximum number of in-flight requests to allow.
+	MaxInFlight = 1000
+	TokenLen    = 24
+)
+
+type requestCache struct {
+	// clock is used to obtain the current time
+	clock clock.Clock
+
+	tokens map[string]*list.Element
+	ll     *list.List
```

Comment:
Yeah, maybe overkill. The logic was copied from [lru](https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/golang/groupcache/lru/lru.go). Added comments.